### PR TITLE
SITE-3473: Add SVAP 2022 USCDI V2 Validator

### PR DESCRIPTION
**Pull request by bgirishrao**

### CODE REVIEW

File: gui/src/app/direct/directHeader.tpl.html [hunk 1]
```diff
@@ -35,12 +35,14 @@
             <li uib-dropdown ng-class="{active: $state.includes('direct.ccda') || $state.includes('direct.ccdar2') || $state.includes('direct.ccdar3')}">
```
Comment (drbgfc) [inline gui/src/app/direct/directHeader.tpl.html:35]: Should there maybe be:
 || $state.includes('direct.ccdasvap2022')
...or something like that? And, is there a CSS class created? And, if not, maybe we don't the ng-class for the other items at all, or those CSS classes?

File: webapp/src/main/java/gov/nist/healthcare/ttt/webapp/common/controller/SVAPValidatorCtrl.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+				}								
+				HttpPost post = new HttpPost(svapUrl);
+				FileBody fileBody = new FileBody(file);
+				//
+				MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+				builder.setMode(HttpMultipartMode.BROWSER_COMPATIBLE);
+				builder.addTextBody("validationObjective", validationObjective);
+				builder.addTextBody("referenceFileName", referenceFileName);
+				builder.addPart("ccdaFile", fileBody);
+				builder.addTextBody("curesUpdate", "true");
```
Comment (drbgfc) [inline webapp/src/main/java/gov/nist/healthcare/ttt/webapp/common/controller/SVAPValidatorCtrl.java:75]: this should be false:
webapp/src/main/java/gov/nist/healthcare/ttt/webapp/common/controller/SVAPValidatorCtrl.java
builder.addTextBody("curesUpdate", "true");

File: webapp/src/main/java/gov/nist/healthcare/ttt/webapp/common/controller/SVAPValidatorCtrl.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
+package gov.nist.healthcare.ttt.webapp.common.controller;
```
Comment (drbgfc) [inline webapp/src/main/java/gov/nist/healthcare/ttt/webapp/common/controller/SVAPValidatorCtrl.java:1]: This is not something that is actiionble now, but, for the future, we really need to think about why we need a backend at all for c-cda validator calls. We should be calling the SITE endpoint directly from the UI if possible. As is, it complicates and obfuscates. I'm sure there is some reason I'm not aware of, but I feel like we can work through it, or if needed, pass the backend work off to the site service.

Comment (bgirishrao) [reply webapp/src/main/java/gov/nist/healthcare/ttt/webapp/common/controller/SVAPValidatorCtrl.java:1]: I think this to to pass the different variable based on the different options

File: gui/src/app/direct/directHeader.tpl.html [hunk 4]
```diff
@@ -32,7 +32,7 @@
                     <div msg="" key="navbar.messageValidator">Message Validator</div>
                 </a>
             </li>
-            <li uib-dropdown ng-class="{active: $state.includes('direct.ccda') || $state.includes('direct.ccdar2') || $state.includes('direct.ccdar3')}">
+            <li uib-dropdown ng-class="{active: $state.includes('direct.ccda') || $state.includes('direct.ccdar2') || $state.includes('direct.ccdar3') || $state.includes('direct.svap})">
```
Comment (drbgfc) [inline gui/src/app/direct/directHeader.tpl.html:35]: -I think you are missing a single-quote after svap
-I think that the } and the ) need to be swapped
Also, I'm curious if the CSS class exists in the CSS? I haven't checked.

File: webapp/src/main/java/gov/nist/healthcare/ttt/webapp/common/controller/SVAPValidatorCtrl.java [hunk 5]
```diff
@@ -72,7 +72,7 @@ public class SVAPValidatorCtrl {
 				builder.addTextBody("validationObjective", validationObjective);
 				builder.addTextBody("referenceFileName", referenceFileName);
 				builder.addPart("ccdaFile", fileBody);
-				builder.addTextBody("curesUpdate", "true");
+				builder.addTextBody("curesUpdate", "false");
```
Comment (drbgfc) [inline webapp/src/main/java/gov/nist/healthcare/ttt/webapp/common/controller/SVAPValidatorCtrl.java:75]: looks good

**Pull request merged**